src/parameters/add_parameters.jl

```diff
@@ -197,6 +197,10 @@ function _add_time_series_parameters!(
     device_names = String[]
     initial_values = Dict{String, AbstractArray}()
     for device in devices
+        if !PSY.has_time_series(device, ts_type, ts_name)
```

Comment:
This is currently slower than you want because it involves a database search. Does this need to be made faster? It's only called once during build, right? If it should be cached, should that be here in PSI or in IS?

Reply:
We also make this calls for some parameter updates. I think we should have a cache for this in IS.